### PR TITLE
Build multiple Android ABIs into one AAR package.

### PR DESCRIPTION
**Pull request by wenbingl**

### CODE REVIEW

**Comment by edgchen1** [comment]
can we add a CI build for this?

**Comment by wenbingl** [comment]
> 

Yes, it will be in the next PR?
Do we have any onnxruntime AAR package test program for reference?

**Comment by edgchen1** [comment]
> Do we have any onnxruntime AAR package test program for reference?

Yeah, here's a test program:
https://github.com/microsoft/onnxruntime/blob/cd2e8b306ca91eaedd50e034995c0e9055cbb079/java/src/test/android/README.md
and usage in the pipeline:
https://github.com/microsoft/onnxruntime/blob/cd2e8b306ca91eaedd50e034995c0e9055cbb079/tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml#L53-L63

**Comment by wenbingl** [comment]
> > Do we have any onnxruntime AAR package test program for reference?
> 
> Yeah, here's a test program: https://github.com/microsoft/onnxruntime/blob/cd2e8b306ca91eaedd50e034995c0e9055cbb079/java/src/test/android/README.md and usage in the pipeline: https://github.com/microsoft/onnxruntime/blob/cd2e8b306ca91eaedd50e034995c0e9055cbb079/tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml#L53-L63

Nice, I will take a look!

File: build.android [hunk 1]
```diff
@@ -5,23 +5,43 @@ set -e -x -u
...
+        -DOCOS_BUILD_ANDROID=ON                                                 \
+        -DANDROID_NDK_VERSION=${CURRENT_NDK_VERSION}                            \
+        ../../.. && cmake --build . --config Release --parallel
+    popd
+done
+
+if [[ ${#ABIS[@]} -gt 1 ]]; then
+    for abi_name in "${ABIS[@]:1}";
+    do
+        cp -R out/$OSNAME/Release_${abi_name}/java/android/${abi_name} out/$OSNAME/Release_x86/java/android/
```
Comment (edgchen1) [inline build.android:42]:
```suggestion
        cp -R out/$OSNAME/Release_${abi_name}/java/android/${abi_name} out/$OSNAME/Release_${ABIS[0]}/java/android/
```

File: build.android [hunk 2]
```diff
@@ -5,23 +5,43 @@ set -e -x -u
...
+    done
+
+    # The final ABI merged package is in out/Release_x86\java\build\android\outputs\aar
+    pushd out/$OSNAME/Release_x86 && cmake --build . --config Release --parallel && popd
```
Comment (edgchen1) [inline build.android:46]:
```suggestion
    pushd out/$OSNAME/Release_${ABIS[0]} && cmake --build . --config Release --parallel && popd
```

File: build.android [hunk 3]
```diff
@@ -5,23 +5,43 @@ set -e -x -u
...
+    popd
+done
+
+if [[ ${#ABIS[@]} -gt 1 ]]; then
+    for abi_name in "${ABIS[@]:1}";
+    do
+        cp -R out/$OSNAME/Release_${abi_name}/java/android/${abi_name} out/$OSNAME/Release_x86/java/android/
+    done
+
+    # The final ABI merged package is in out/Release_x86\java\build\android\outputs\aar
```
Comment (edgchen1) [inline build.android:45]:
```suggestion
    # The final ABI merged package is in out/${OSNAME}/Release_${ABIS[0]}/java/build/android/outputs/aar
```

File: build.android (hunk 2 above)
Comment (edgchen1) [inline build.android:46]: perhaps I'm missing something, why do we build in this directory again?

Comment (wenbingl) [reply build.android]: accepted.

File: build.android (hunk 3 above)
Comment (wenbingl) [reply build.android]: accepted.

File: build.android (hunk 2 above)
Comment (wenbingl) [reply build.android]: After other ABI JNI libraries are ready, gradle needs to be rerun to generate the final package. To be simple here, we just call cmake here to invoke gradle.

File: build.android [hunk 4]
```diff
@@ -5,23 +5,45 @@ set -e -x -u
...
+    for abi_name in "${ABIS[@]:1}";
+    do
+        cp -R out/$OSNAME/Release_${abi_name}/java/android/${abi_name} out/$OSNAME/Release_${ABIS[0]}/java/android/
+    done
+
+    # The final ABI merged package is in out/Release_${ABIS[0]}\java\build\android\outputs\aar
+    pushd out/$OSNAME/Release_${ABIS[0]} && cmake --build . --config Release --parallel && popd
+fi
+
+echo "The final android AAR package is located at: out/Release_${ABIS[0]}/java/build/android/outputs/aar"
```
Comment (edgchen1) [inline build.android:49]:
```suggestion
echo "The final android AAR package is located at: out/${OSNAME}/Release_${ABIS[0]}/java/build/android/outputs/aar"
```

nit: could check for ABIS not being empty

**Review by edgchen1 — APPROVED** [review]

File: build.android (hunk 4 above)
Comment (wenbingl) [reply build.android]: fixed, good catch!

**Pull request merged**